src/main/webapp/app/exercises/modeling/assess/modeling-assessment-editor/modeling-assessment-dashboard.component.html

```diff
@@ -167,86 +170,128 @@ <h5 *ngIf="modelingExercise.assessmentType === AssessmentType.SEMI_AUTOMATIC &&
-                            [routerLink]="[
-                                '/course-management',
-                                modelingExercise.course?.id,
-                                'modeling-exercises',
-                                modelingExercise.id,
-                                'submissions',
-                                submission.id,
-                                'assessment'
-                            ]"
-                            [disabled]="busy"
-                            [queryParams]="{ optimal: false, hideBackButton: true }"
-                            class="btn btn-outline-secondary btn-sm mb-1"
-                        >
-                            <fa-icon [icon]="'folder-open'" [fixedWidth]="true"></fa-icon>&nbsp;{{ 'artemisApp.assessment.dashboard.actions.open' | translate }}
-                        </button>
-                        <button
-                            *ngIf="modelingExercise.type === ExerciseType.MODELING && submission && submission.latestResult && modelingExercise.exerciseGroup"
-                            [routerLink]="[
-                                '/course-management',
-                                modelingExercise.exerciseGroup?.exam.course.id,
-                                'modeling-exercises',
-                                modelingExercise.id,
-                                'submissions',
-                                submission.id,
-                                'assessment'
-                            ]"
-                            [disabled]="busy"
-                            [queryParams]="{ optimal: false, hideBackButton: true, 'correction-round': submission.results?.length }"
-                            class="btn btn-outline-secondary btn-sm mb-1"
-                        >
-                            <fa-icon [icon]="'folder-open'" [fixedWidth]="true"></fa-icon>&nbsp;{{ 'artemisApp.assessment.dashboard.actions.open' | translate }}
-                        </button>
-                        <span *ngIf="modelingExercise.isAtLeastInstructor">
-                            <button
-                                *ngIf="submission && submission.latestResult && !submission.latestResult!.completionDate"
-                                (click)="cancelAssessment(submission)"
-                                [disabled]="busy"
-                                class="btn btn-outline-secondary btn-sm mb-1"
-                            >
-                                <fa-icon [icon]="'ban'" [fixedWidth]="true"></fa-icon>&nbsp;{{ 'artemisApp.assessment.dashboard.actions.cancel' | translate }}
-                            </button>
+                            <!-- exercise without exam-->
+                            <span *ngIf="modelingExercise.type === ExerciseType.MODELING && submission && modelingExercise.course">
```

Comment:
That if statement is big and used in a couple of places. You can technically extract that into an `isExerciseWithoutExam()` function

Reply:
I don't think that's necessary here, as it does not simplify the html a lot.